### ISSUE TITLE
Use custom validation set for training

### ISSUE THREAD

**Issue by anwai98** [MIC-DKFZ/nnUNet#2060]
Hi team,

I'd want to train nnUNet, on the train set (formatted at: `imagesTr, labelsTr`) and my custom validation set (formatted at `imagesVal, labelsVal`). Could you guide me in the direction of how can this be done? (maybe I am missing passing an argument to automatically take care of it, or need to add this somewhere in `dataset.json`)

PS: I have some thoughts on how this might be done, just want to be super sure if my custom validation set is being used:
1. preprocess the (training) images (how should I also enable preprocessing for the val split)
2. train using `nnUNetv2_train <DATASET_ID> all`

Really nice and modular update with nnUNetv2!
Thanks in advance! :)

**Comment by anwai98** [comment]
Just to keep posted in case someone stumbles on this, it's possible to create custom splits at ease using the [documentation](https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/manual_data_splits.md) here (thanks for the well-documented features ;) )

**Comment by TSbme** [comment]
> Just to keep posted in case someone stumbles on this, it's possible to create custom splits at ease using the [documentation](https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/manual_data_splits.md) here (thanks for the well-documented features ;) )
Hi, I noticed that 'fold' is a required parameter option. After fixing the training and validation sets, is the training command still ‘nnUNetv2_train <DATASET_ID> all’?

**Comment by anwai98** [comment]
Hi @TSbme,

> Hi, I noticed that 'fold' is a required parameter option. After fixing the training and validation sets, is the training command still ‘nnUNetv2_train <DATASET_ID> all’?

Answering your question on my mention, the way I went ahead with this is:
- Placing the train and val datasets in `imagesTr` (for all images) and `labelsTr` (for all ground-truth)
- Creating a custom `splits_final.json` file in `nnUNet_preprocessed/DatasetXYZ` by specific mentions of train and val images' file names in the respective splits
- Run the training for only one fold, so training (and later inferring) for only fold 0 should do the trick (it's not really needed to train 5 times as you have a custom specified validation set, in contrast to the cross-validation setup supported by nnUNet)

**Comment by TSbme** [comment]
@anwai98 Yes, thank you!

**Comment by mrokuss** [comment]
Hey @anwai98 

In your last post you described exactly the correct way! Closing.
